pkg/networkmanager/firewall.jsx

```diff
@@ -171,6 +193,136 @@ class AddServicesBody extends React.Component {
         });
     }
 
+    parse_services(content) {
+        if (!content) {
+            console.warn("Couldn't read /etc/services");
+            return [ ];
+        }
+
+        var lines = content.split('\n');
+        var ret = {};
+
+        for (var i = 0; i < lines.length; i++) {
+            if (!lines[i] || lines[i].startsWith("#"))
+                continue;
+            var m = lines[i].match(/^(\S+)\s+(\d+)\/(\S+).*?(#(.*))?$/);
```

Comment:
Note for self: `services(5)` actually says "The  service-name  should  begin in the first column of the file, since leading spaces are not stripped.", so this is okay.

However, please use `let` or `const` in any block that isn't directly a `function` block (and it's fine to use `let` there as well), due to the broken scoping of `var`.